src/SingleLineTextEdit.cpp

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "Host.h"
+#include "SingleLineTextEdit.h"
+#include <QKeyEvent>
```

Comment:
We do need a couple of helpers around the inclusion of any Qt headers - for MSVC compilation (not that we currently do any but @ahmedcharles put support for that in years ago and we {cargo-cult?} continue to add it where required) - so check for it on all your new files please...
```suggestion
#include "pre_guard.h"
#include <QKeyEvent>
#include "post_guard.h"
```

Reply:
roger that, done